### PR TITLE
[2038] Fix stock round corporation grouping

### PR DESCRIPTION
**Pull request by jason-johnson**
In the standard game only "Group A" corporations are initially available.  After all corporations in that group IPO the next group becomes available.  The same repeats again until the final group C become available.

Also fixed the corporation colours to match the real game.

### CODE REVIEW

File: lib/engine/game/g_2038/game.rb [hunk 1]
```diff
@@ -540,7 +540,7 @@ class Game < Game::Base
             simple_logo: '1830/PRR.alt',
             tokens: [60, 100, 60, 100, 60, 100, 60, 100, 60, 100],
             coordinates: 'K9',
-            color: '#32763f',
+            color: '#40b1b9',
             type: 'groupA',
```
Comment (tobymao) [inline lib/engine/game/g_2038/game.rb:544]: change to snake_case

File: lib/engine/game/g_2038/game.rb [hunk 2]
```diff
@@ -652,6 +652,52 @@ def game_hexes
...
+
+          @b_group_corporations, @c_group_corporations = @b_group_corporations.partition do |corporation|
+            corporation.type == :groupB
+          end
+        end
+
+        def event_group_b_corps_available!
+          @log << 'Group B corporations are now available'
+
+          @corporations += @b_group_corporations
```
Comment (tobymao) [inline lib/engine/game/g_2038/game.rb:682]: @corporations <<

File: lib/engine/game/g_2038/game.rb [hunk 3]
```diff
@@ -652,6 +652,52 @@ def game_hexes
...
+          @b_group_corporations = []
+          @available_corp_group = :groupB
+        end
+
+        def event_group_c_corps_available!
+          @log << 'Group C corporations are now available'
+
+          @corporations += @c_group_corporations
```
Comment (tobymao) [inline lib/engine/game/g_2038/game.rb:690]: same?

File: lib/engine/game/g_2038/game.rb (hunk 2 above)
Comment (jason-johnson) [reply lib/engine/game/g_2038/game.rb:682]: @b_group_corporations is an array, I wouldn't want to append that, not insert it, right?

Comment (tobymao) [reply lib/engine/game/g_2038/game.rb:682]: ok, then do concat

**Review by tobymao — APPROVED** [review]

**Pull request merged**